src/LocalTranslationsBatcher.php

```diff
@@ -151,7 +158,30 @@ protected function getExtensionsToTranslate(): array {
       $extensions[] = $name;
     }
 
-    return $extensions;
+    return array_intersect_key(locale_translation_project_list(), array_combine($extensions, $extensions));
+  }
+
+  /**
+   * Implements callback_batch_operation().
+   *
+   * Import po files of the project for allowed languages.
+   */
+  public static function importProjectPoFiles($extension, $langcodes, &$context): void {
+    $files = file_scan_directory(
+      \Drupal::service('file_system')->dirname($extension['info']['interface translation server pattern']),
+      '/.*-(' . implode('|', $langcodes) . ')\.po/'
```

Comment:
Any reason not to inject the `file_system` service?

Reply:
This is static method.